hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyDeleteRequest.java

```diff
@@ -126,6 +126,8 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       if (omKeyInfo == null) {
         throw new OMException("Key not found", KEY_NOT_FOUND);
       }
+      // Set the UpdateID to current transactionLogIndex
+      omKeyInfo.setUpdateID(transactionLogIndex);
```

Comment:
I think, we don't need to set updateID, as we delete this entry from KeyTable anyway. (As we set Optional.absent() below, which will be deleted from OM DB eventually when double buffer picks). Or is there any scenario by setting this helps.

Example: Create key Key1 - 1, Delete Key Key1 - 2 Create Key Key1 (again) - 3 Now replay happening from 1, as now in KeyTable it will have updateID 3, we think it as replay just by checking OmKeyInfo.getUpdateID, similar to Bucket/Volume delete.

Reply:
Deleted Keys are put in DeletedKey Table and then purged when we get KeyPurgeRequest. As the purge request could also be a replay, we should only delete the keys which were deleted before the purge request. For this check, we need to set the updateID here.